hoodie-client/src/main/java/com/uber/hoodie/io/HoodieCommitArchiver.java

```diff
@@ -0,0 +1,273 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.hoodie.io;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Sets;
+import com.uber.hoodie.avro.model.ActionType;
+import com.uber.hoodie.avro.model.HoodieCleanMetadata;
+import com.uber.hoodie.avro.model.HoodieMetadataWrapper;
+import com.uber.hoodie.avro.model.HoodieRollbackMetadata;
+import com.uber.hoodie.avro.model.HoodieSavepointMetadata;
+import com.uber.hoodie.common.model.HoodieCommitMetadata;
+import com.uber.hoodie.common.model.HoodieCompactionMetadata;
+import com.uber.hoodie.common.table.HoodieTableMetaClient;
+import com.uber.hoodie.common.table.HoodieTimeline;
+import com.uber.hoodie.common.table.log.HoodieLogFile;
+import com.uber.hoodie.common.table.log.HoodieLogFormat;
+import com.uber.hoodie.common.table.log.block.HoodieAvroDataBlock;
+import com.uber.hoodie.common.table.timeline.HoodieArchivedTimeline;
+import com.uber.hoodie.common.table.timeline.HoodieInstant;
+import com.uber.hoodie.common.util.AvroUtils;
+import com.uber.hoodie.config.HoodieWriteConfig;
+import com.uber.hoodie.exception.HoodieCommitException;
+import com.uber.hoodie.exception.HoodieException;
+import com.uber.hoodie.exception.HoodieIOException;
+import com.uber.hoodie.table.HoodieTable;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.IndexedRecord;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Archiver to bound the growth of <action>.commit files
+ */
+public class HoodieCommitArchiver {
+    private static Logger log = LogManager.getLogger(HoodieCommitArchiver.class);
+
+    public static String dummyCommitTimeForArchive = "0";
+    private final Path archiveFilePath;
+    private final FileSystem fs;
+    private final HoodieWriteConfig config;
+    private HoodieLogFormat.Writer writer;
+
+    public HoodieCommitArchiver(HoodieWriteConfig config, FileSystem fs) {
+        this.fs = fs;
+        this.config = config;
+        this.archiveFilePath = HoodieArchivedTimeline
+            .getArchiveLogPath(config.getBasePath() + "/" + HoodieTableMetaClient.METAFOLDER_NAME);
+    }
+
+    private HoodieLogFormat.Writer openWriter(String commitTime) {
+        try {
+            return HoodieLogFormat.newWriterBuilder()
+                    .onParentPath(new Path(config.getBasePath()))
+                    .withFileId(archiveFilePath.getName())
+                    .withFileExtension(HoodieLogFile.DELTA_EXTENSION)
+                    .withFs(fs)
+                    .overBaseCommit(commitTime).build();
```

Comment:
Passing an empty commitTime will lead to a file name of the format : .commits.archived_.log.1. Having an "_" followed by an empty string looks really odd hence added the dummyCommitTime. 
We do need to pass a commitTime because of this : https://github.com/uber/hoodie/blob/master/hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieLogFormat.java#L149. It does make sense to make commitTime required in the HoodieLogFormat. 
Do you think "commits.archived_.log.1" is a good file name ? That "_" really irks me.